src/axom/lumberjack/tests/CMakeLists.txt

```diff
@@ -7,65 +7,62 @@
 #------------------------------------------------------------------------------
 
 #------------------------------------------------------------------------------
-# Specify list of tests
+# Add serial GTests based tests
 #------------------------------------------------------------------------------
+set(lumberjack_serial_tests
+    lumberjack_Lumberjack.hpp
+    lumberjack_Message.hpp
+    lumberjack_TextEqualityCombiner.hpp )
 
-# Add Serial GTests based tests
-set(gtest_lumberjack_tests
-    lumberjack_Lumberjack.cpp
-    lumberjack_Message.cpp
-    lumberjack_TextEqualityCombiner.cpp
-    )
+blt_add_executable(NAME       lumberjack_serial_tests
+                   SOURCES    lumberjack_serial_main.cpp ${lumberjack_serial_tests}
+                   OUTPUT_DIR ${TEST_OUTPUT_DIRECTORY}
+                   DEPENDS_ON lumberjack gtest 
+                   FOLDER     axom/lumberjack/tests )
 
-foreach(test ${gtest_lumberjack_tests})
-    get_filename_component( test_name ${test} NAME_WE )
-    blt_add_executable( NAME ${test_name}_test
-                        SOURCES ${test}
-                        OUTPUT_DIR ${TEST_OUTPUT_DIRECTORY}
-                        DEPENDS_ON lumberjack gtest 
-                        FOLDER axom/lumberjack/tests )
-    blt_add_test( NAME ${test_name} 
-                  COMMAND ${test_name}_test )
+foreach(test_suite ${lumberjack_serial_tests})
+    get_filename_component(test_suite ${test_suite} NAME_WE)
+    blt_add_test(NAME    ${test_suite}
+                 COMMAND lumberjack_serial_tests --gtest_filter=${test_suite}* )
```

Comment:
What does the ``--gtest_filter`` do?

Reply:
It only runs the tests that meet the regex i gave it.  for this example it would be the filename without the .hpp.

Googletest uses this format to name things: [test suite].[test name]

I follow the convention:
test suite =  [component name]_[class name]

then test name is more specific to what I'm testing

So for example:

lumberjack_BinaryCommunicator.pushNothing